src/Sulu/Component/PHPCR/PathCleanup.php

```diff
@@ -11,8 +11,13 @@
 
 namespace Sulu\Component\PHPCR;
 
+use Symfony\Component\String\Slugger\AsciiSlugger;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
 /**
  * cleans path strings.
+ *
+ * @internal this class is for internal use
```

Comment:
I would just write `@internal`, since the additional comment is just an explanation of what internal means.

Reply:
Or better say if you use it - use it throught services provided by bundle?

Reply:
If we allow that the service/class is not internal IMO. Maybe that's the even better answer to this? Other people are already using this service, so there seems to be a need for it. Maybe it should not be internal at all?

Reply:
Yup, I think it can be used, but not instantiate it directly. That was the main issue here. 
People should follow the interface, not class - but in this case we DIDN'T changed the class because people don't follow  the interface. (The interface does not contains anything about __construct)

Reply:
If people follows the inteface, they should care only about `cleanup` and `validate` methods to be same - and this would not  be necessary

`        if (null === $slugger) {
            @trigger_error(
                'Initializing the PathCleanup without a slugger is deprecated since Sulu 2.1.',
                E_USER_DEPRECATED
            );
            $slugger = new AsciiSlugger();
        }`

Reply:
That's at least not my understanding of the `@internal` tag. I would assume that the entire class may not be used outside of the sulu/sulu package. Let me quote the [phpDocumentor documentation](https://docs.phpdoc.org/latest/references/phpdoc/tags/internal.html#description):

> The @internal tag can be used as counterpart of the @api tag, indicating that the associated Structural Elements are used purely for the internal workings of this piece of software.

@sulu/core-team Does anyone of you have a different opinion on that? If this annotation is not used consistently they are not very helpful.

Reply:
Thats my understanding of the `@internal` tag too, therefore i would not mark the `PathCleanup` as internal. 

I think it is an anti-pattern in general to directly instantiate something that is available as a service. So i dont think it is necessary to explicitly clarify this here. 🙂